hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestSlowLogRecorder.java

```diff
@@ -140,12 +141,16 @@ public void testOnlieSlowLogConsumption() throws Exception {
     Assert.assertNotEquals(-1, HBASE_TESTING_UTILITY.waitFor(3000,
       () -> slowLogRecorder.getSlowLogPayloads(request).size() == 7));
 
-    slowLogPayloads = slowLogRecorder.getSlowLogPayloads(request);
-
-    Assert.assertEquals(slowLogPayloads.size(), 7);
-    confirmPayloadParams(0, 7, slowLogPayloads);
-    confirmPayloadParams(5, 2, slowLogPayloads);
-    confirmPayloadParams(6, 1, slowLogPayloads);
+    Assert.assertNotEquals(-1, HBASE_TESTING_UTILITY.waitFor(3000,
+      () -> {
+        List<SlowLogPayload> slowLogPayloadsList = slowLogRecorder.getSlowLogPayloads(request);
+        Assert.assertEquals(slowLogPayloadsList.size(), 7);
+        boolean b1 = confirmPayloadParams(0, 7, slowLogPayloadsList);
```

Comment:
I don't understand why these tests are executed within the timeout window. How does that make the tests more reliable?

Reply:
Since we are dealing with parallel processing and we are expecting consumer of ring buffer to get done with ordered events, but based on env, consumer might not be done by the time we expect ordered events present and hence we want to introduce time lag until we get that specific end result of expected ordered list of events in ring buffer. Sounds good @ndimiduk  ?

Reply:
e.g if we send {1,2,3,4,5,6,7} to RingBuffer of size 8, then all of them are consumed in same order. Now if we send {8,9,10,11}, we expect final output from RingBuffer consumer to be {9,10,11,4,5,6,7,8}  (1,2,3 are overridden). However, the reason why we have flakes is because by the time we expect above output, consumer might not have consumed say 10 and 11 and hence actual output would be {9,2,3,4,5,6,7,8}. And hence the failures, so now with waitFor(), we wait until we get ordered output {9,10,11,4,5,6,7,8}, which is why we will wait for all of 8,9,10,11 to be consumed and put in the queue in correct order i.e. no flakes.

Reply:
Oh, I see. So you're actually using `waitFor` as a retry mechanism. That's not how I imagined using it, but so long as those inner asserts don't interfere with the retry, I guess that's fine.

What happens when the inner `Assert.assertEquals(slowLogPayloadsList.size(), 7);` throws? Does `waitFor` retry that, or does it simply fail? If the latter, the test is still flakey. I would loop until I have all the data and then perform the asserts at the end.

Reply:
oops, my bad, I missed `Assert.assertEquals(slowLogPayloadsList.size(), 7);`, this still makes it flakey as it fails. Let me take care of it in Addendum. 
Thanks @ndimiduk